src/servers/azure.ts

```diff
@@ -0,0 +1,48 @@
+import Server, { RequestHandler } from '../server';
+// @ts-ignore
+import type { Context, HttpRequest } from '@azure/functions';
+
+/**
+ * A server for Azure Function integration
+ * @see https://docs.microsoft.com/en-us/azure/azure-functions/
+ */
+class AzureFunctionServer extends Server {
+  private _handler?: RequestHandler;
+
+  constructor(moduleExports: any, target = 'interactions') {
+    super({ alreadyListening: true });
+    moduleExports[target] = this._onRequest.bind(this);
+  }
+
+  private _onRequest(context: Context, req: HttpRequest) {
+    if (!this._handler) {
+      context.res!.status = 503;
+      context.res!.send('Server has no handler');
+    }
+    if (req.method !== 'POST') {
+      context.res!.status = 400;
+      context.res!.send('Server only supports POST requests.');
+    }
+    this._handler!(
+      {
+        headers: req.headers,
+        body: req.body,
+        request: req,
+        response: context.res
+      },
+      async (response) => {
+        context.res!.status = response?.status ?? 200;
```

Comment:
This is failing tests.

Reply:
All tests are passing on my side. 
I thinki it's because I use node 14 on my local copy, and CI is testing with Node 12. Optional chaining must still be behind a feature flag in v12.
I'll change node runtime to v14 in test.yml.
If using v14 is a problem, I can also use an older syntax, that's just sugar coating.